### ISSUE TITLE
implicit declaration of function 'htobe16'

### ISSUE THREAD

**Issue by yunusemreayhan** [intel/tpm2-tss#706]
```
  CC     marshal/base-types.lo
cc1: warnings being treated as errors
marshal/base-types.c: In function 'Tss2_MU_BYTE_Marshal':
marshal/base-types.c:176: error: implicit declaration of function 'htobe16'
marshal/base-types.c:176: error: implicit declaration of function 'htobe32'
marshal/base-types.c:176: error: implicit declaration of function 'htobe64'
marshal/base-types.c: In function 'Tss2_MU_BYTE_Unmarshal':
marshal/base-types.c:177: error: implicit declaration of function 'be16toh'
marshal/base-types.c:177: error: implicit declaration of function 'be32toh'
marshal/base-types.c:177: error: implicit declaration of function 'be64toh'
```
Functions like htole64, htole32 were added to glibc in version 2.9, while mine is older i guess. How can we solve this?

Thx in advance.

**Comment by flihp** [comment]
`i guess` isn't much to go on. Are you really running a system with an 8 year old version of glibc? Hard to help you debug this w/o an accurate version number.

**Comment by flihp** [comment]
A quick scan over the man page for `htobe*` on my dev box shows that it's more likely that you have a glibc version between 2.9 and 2.19. In this case you'll need to use a different feature test macro. Try adding `_BSD_SOURCE` to your build and let us know if that fixes the issue. We can add this macro alongside the `_DEFAULT_SOURCE` currently used (for `glibc >= 2.19`)

**Comment by yunusemreayhan** [comment]
2.26 is the version I am using. Yes, I have an old product, trying to implement this. I guess I will update that package.

**Comment by flihp** [comment]
2.26 is the latest version isn't it? 2.26 was just released ~5 months ago. My development system is still on 2.24. What distro are you using?

**Comment by flihp** [comment]
Any chance this issue was resolved by abc63b38e2ec8ebbbeb14c5b379362742cd088a0?

**Comment by yunusemreayhan** [comment]
yes, I have checked that it compiles now. thanks.
